domain/schema/model/sql/0015-charm.sql

```diff
@@ -68,6 +64,45 @@ CREATE TABLE charm (
 CREATE UNIQUE INDEX idx_charm_reference_name_revision
 ON charm (reference_name, revision);
 
+
+CREATE TABLE charm_provenance (
+    id INT PRIMARY KEY,
+    name TEXT NOT NULL
+);
+
+CREATE UNIQUE INDEX idx_charm_provenance_name
+ON charm_provenance (name);
+
+-- The provenance of the charm. This is used to determine where the charm
+-- came from, which can then determine if the download information is still
+-- relevant.
+INSERT INTO charm_provenance VALUES
+(0, 'download'),
```

Comment:
When you are deploying a charm on a model, is `download` the provenance that will be used?

Reply:
`download` is used for server-side deployments (pass in a charmhub name) and in set charm for non-server-side deployments.